element_calcium_imaging/export/nwb/nwb.py

```diff
@@ -0,0 +1,331 @@
+    """
+
+    imaging_files = (scan.ScanInfo.ScanFile & session_key).fetch("file_path")
+    two_p_series = TwoPhotonSeries(
+        name="TwoPhotonSeries",
+        dimension=(scan.ScanInfo.Field & session_key).fetch1("px_height", "px_width"),
+        external_file=imaging_files,
+        imaging_plane=imaging_plane,
+        starting_frame=[0],
+        format="external",
+        starting_time=0.0,
+        rate=(scan.ScanInfo & session_key).fetch1("fps"),
+    )
+    return two_p_series
+
+
+def _add_motion_correction_to_nwb(session_key, nwbfile):
+    raise NotImplementedError(
+        "Motion Correction data cannot be packaged into NWB at this time."
+    )
+
+
+def _add_segmentation_data_to_nwb(session_key, nwbfile, imaging_plane):
+    """Adds segmentation data from database.
+
+    Args:
+        session_key (dict): key from Session table
+        nwbfile (NWBFile): nwb file
+        imaging_plane (NWBFile Imaging Plane): nwb file imaging plane object
+    """
+
+    ophys_module = nwbfile.create_processing_module(
+        name="ophys", description="optical physiology processed data"
+    )
+    img_seg = ImageSegmentation()
+    ps = img_seg.create_plane_segmentation(
+        name="PlaneSegmentation",
+        description="output from segmenting",
+        imaging_plane=imaging_plane,
+    )
+    ophys_module.add(img_seg)
+
+    mask_keys = (imaging.Segmentation.Mask & session_key).fetch("KEY")
+    for mask_key in mask_keys:
+        ps.add_roi(
+            pixel_mask=np.asarray(
+                (imaging.Segmentation.Mask() & mask_key).fetch1(
+                    "mask_xpix", "mask_ypix", "mask_weights"
+                )
+            ).T
+        )
+
+    rt_region = ps.create_roi_table_region(
+        region=((imaging.Segmentation.Mask & session_key).fetch("mask")).tolist(),
+        description="All ROIs from database.",
+    )
+
+    channels = (scan.ScanInfo & session_key).fetch1("nchannels")
+    for channel in range(channels):
+        roi_resp_series = RoiResponseSeries(
+            name=f"Fluorescence_{channel}",
+            data=np.stack(
+                (
+                    imaging.Fluorescence.Trace
+                    & session_key
+                    & f"fluo_channel='{channel}'"
+                ).fetch("fluorescence")
+            ).T,
+            rois=rt_region,
+            unit="a.u.",
+            rate=(scan.ScanInfo & session_key).fetch1("fps"),
+        )
+        neuropil_series = RoiResponseSeries(
+            name=f"Neuropil_{channel}",
+            data=np.stack(
+                (
+                    imaging.Fluorescence.Trace
+                    & session_key
+                    & f"fluo_channel='{channel}'"
+                ).fetch("neuropil_fluorescence")
+            ).T,
+            rois=rt_region,
+            unit="a.u.",
+            rate=(scan.ScanInfo & session_key).fetch1("fps"),
+        )
+        # deconvolved_series = RoiResponseSeries(
```

Comment:
@kushalbakshi is this being commented out by mistake?

Reply:
Yes it was commented as I was trying different things while testing. It has been reverted in the most recent commit